### PR TITLE
MapEntries IR; use in annotate.

### PR DESCRIPTION
**Pull request by catoverdrive**
I'm going to merge select/drop/annotate on the scala side in a separate PR. For now annotateEntriesExpr uses the IR path if possible for testing.

### CODE REVIEW

File: python/hail/tests/utils.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 
 def startTestHailContext():
-    hail.init(master='local[2]', min_block_size=0, quiet=True)
+    hail.init(master='local[2]', min_block_size=0)
```
Comment (tpoterba) [inline python/hail/tests/utils.py]: unintentional change

File: src/main/scala/is/hail/expr/Relational.scala [hunk 2]
```diff
@@ -379,6 +379,76 @@ case class FilterRows(
...
+    InsertFields(Ref("va"), Seq((MatrixType.entriesIdentifier, entries)))
+  }
+
+  val typ: MatrixType = {
+    Infer(newRow, None, new Env[Type]()
+      .bind("global", child.typ.globalType)
+      .bind("va", child.typ.rvRowType)
+      .bind("sa", TArray(child.typ.colType))
+    )
+    child.typ.copy(rvRowType=newRow.typ)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/Relational.scala]: style - should have spaces

File: src/main/scala/is/hail/expr/Relational.scala [hunk 3]
```diff
@@ -379,6 +379,76 @@ case class FilterRows(
...
+        val region = rv.region
+        val oldRow = rv.offset
+
+        rvb.set(region)
+        rvb.start(localGlobalsType)
+        rvb.addAnnotation(localGlobalsType, globalsBc.value)
+        val globals = rvb.end()
+
+        rvb.start(localColsType)
+        rvb.addAnnotation(localColsType, colValuesBc.value)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/Relational.scala]: prefer addUnsafeArray here

Comment (tpoterba) [reply src/main/scala/is/hail/expr/Relational.scala]: and :( this is going to be so slow, the off-heap changes can't come fast enough!

Comment (tpoterba) [reply src/main/scala/is/hail/expr/Relational.scala]: I'm having a bit of trouble reasoning about this. `rv.region` is usually the same region for every record, right? Is this going to mean that the sample annotations are added to the region once for every record in a partition?

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/Relational.scala]: :( yes. But we can't get around that right now since we're not creating the original region.

Comment (tpoterba) [reply src/main/scala/is/hail/expr/Relational.scala]: this might be untenable, even in the short term. Let's discuss at checkin

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/Relational.scala]: re: addUnsafeArray---I kind of prefer the consistency of addAnnotation. It calls addUnsafeArray in the UnsafeIndexedSeq case anyways.

**Pull request merged**